solutions/src/main/java/ru/sberbank/school/task02/FxClient.java

```diff
@@ -0,0 +1,118 @@
+package ru.sberbank.school.task02;
+
+import lombok.NonNull;
+import ru.sberbank.school.task02.exception.FxConversionException;
+import ru.sberbank.school.task02.exception.NullQuotesListException;
+import ru.sberbank.school.task02.exception.WrongSymbolException;
+import ru.sberbank.school.task02.util.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Класс-реализация клиента для получения расчетных курсов
+ */
+public class FxClient implements FxClientController {
+
+    /**
+     * Текущий конвертер. Инициализируется через конструктор
+     */
+    private FxConversionService currentConverter;
+
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss");
+
+
+    FxClient(FxConversionService converter) {
+        currentConverter = converter;
+    }
+
+    /**
+     * Функция обрабатывает список запросов и возвращает список ответов
+     *
+     * @param requests список запросов
+     * @throws IllegalArgumentException если amount не является числом, либо amount <= 0
+     *                                  если операция задана неправильно
+     * @throws NullPointerException     если лист запросов null
+     *                                  если какой-либо из запросов null
+     *                                  если какая-либо составляющая запросов null
+     * @throws FxConversionException    если лист котировок пуст
+     * @throws WrongSymbolException     если symbol не поддерживается
+     */
+    @Override
+    public List<FxResponse> fetchResult(@NonNull List<FxRequest> requests) {
+        List<FxResponse> responses = new ArrayList<>();
+        for (FxRequest req : requests) {
+            responses.add(fetchResult(req));
+        }
+        return responses;
+    }
+
+
+    /**
```

Comment:
First sentence of Javadoc is missing an ending period.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1978380870>*